apps/prairielearn/src/pages/instructorInstanceAdminPublishing/components/LegacyAccessRuleCard.tsx

```diff
@@ -0,0 +1,108 @@
+import { formatDate } from '@prairielearn/formatter';
+
+import { CommentPopover } from '../../../components/CommentPopover.js';
+import type { CourseInstance, CourseInstanceAccessRule } from '../../../lib/db-types.js';
+
+export function LegacyAccessRuleCard({
+  accessRules,
+  showComments,
+  courseInstance,
+  hasCourseInstancePermissionView,
+}: {
+  accessRules: CourseInstanceAccessRule[];
+  showComments: boolean;
+  courseInstance: CourseInstance;
+  hasCourseInstancePermissionView: boolean;
+}) {
+  return (
+    <div class="card mb-4">
+      <div class="card-header bg-primary text-white d-flex align-items-center justify-content-between">
+        <h1>Access rules</h1>
+      </div>
+      <div class="alert alert-warning" role="alert">
+        <strong>Legacy Access Rules Active:</strong> This course instance is using the legacy
+        allowAccess system. To use the new access control system, you must first remove all
+        allowAccess rules from the course configuration.
+      </div>
+
+      <div class="table-responsive">
+        <table class="table table-sm table-hover" aria-label="Access rules">
+          <thead>
+            <tr>
+              {showComments && (
+                <th style="width: 1%">
+                  <span class="visually-hidden">Comments</span>
+                </th>
+              )}
+              <th>UIDs</th>
+              <th>Start date</th>
+              <th>End date</th>
+              <th>Institution</th>
+            </tr>
+          </thead>
+          <tbody>
+            {accessRules.map((accessRule) => (
+              <AccessRuleRow
+                key={accessRule.id}
+                accessRule={accessRule}
+                timeZone={courseInstance.display_timezone}
+                hasCourseInstancePermissionView={hasCourseInstancePermissionView}
+                showComments={showComments}
+              />
+            ))}
+          </tbody>
+        </table>
+      </div>
+    </div>
+  );
+}
+
+function AccessRuleRow({
+  accessRule,
+  timeZone,
+  hasCourseInstancePermissionView,
+  showComments,
+}: {
+  accessRule: CourseInstanceAccessRule;
+  timeZone: string;
+  hasCourseInstancePermissionView: boolean;
+  showComments: boolean;
+}) {
+  return (
+    <tr>
+      {showComments && (
+        <td>
+          <CommentPopover comment={accessRule.json_comment} />
+        </td>
+      )}
+      <td>
+        {accessRule.uids == null ? (
+          '—'
+        ) : // Only users with permission to view student data are allowed to
```

Comment:
Nit, formatting is funky - maybe `run(...)` + early return would help? Would also IMO be clearer than chained ternaries.

Reply:
Fixed in 55c48160a88c191bdb71a899849ab38849c3267c.